### PR TITLE
Compare encoded bodies instead of data when possible Fixes #23

### LINKED ISSUE #23: Request HTTPBody serialized is different from the one being deserialized.
This is because json encodes and decodes parameters in a non-deterministic order. This causes tests to fail.

Therefore we cannot directly compare `HTTPBody`'s like we do here:
https://github.com/venmo/DVR/blob/master/DVR/Cassette.swift#L17

### PR DESCRIPTION
**Pull request by dasmer**
We now unwrap the `HTTPBody` `NSData`s into and compare using `isEqual`

### CODE REVIEW

File: DVR/Cassette.swift [hunk 1]
```diff
@@ -13,8 +13,20 @@ struct Cassette {
         for interaction in interactions {
             let r = interaction.request
 
+            let equivalentBody: Bool
+
+            if let rBody = r.HTTPBody,
```
Comment (eliperkins) [inline DVR/Cassette.swift]: Can we avoid the shortened vars here and just call this `requestBody`?

Comment (eliperkins) [inline DVR/Cassette.swift]: Oh, derp. Just noticed we already have `requestBody`... I'm just confused by the naming. What is `r` vs. what is `request`?

**Comment by eliperkins** [comment]
Nice fix! One question around variable naming and then :green_apple:

File: DVR/Cassette.swift (hunk 1 above)
Comment (ayanonagon) [inline DVR/Cassette.swift]: Maybe `interactionRequestBody`?

**Comment by dasmer** [comment]
changed the naming and moved the HTTPBody equality check into into a private function

**Comment by eliperkins** [comment]
LGTM. :green_apple:

**Comment by lionel-alves** [comment]
One question then :green_apple:!

**Pull request merged**